source/objects/RenderTarget.cpp

```diff
@@ -209,6 +217,174 @@ Pixel RenderTarget::SampleBilinear(float u, float v, uint8_t mip) const
 	};
 }
 
+bool RenderTarget::Load(const char* filepath, bool generateMips, bool scaleToRT)
+{
+	if (!IsValid()) return false;
+
+	int imageWidth = 0;
+	int imageHeight = 0;
+	int channels = CHANNELS[static_cast<uint8_t>(GetFormat())];
+	size_t stride = STRIDES[static_cast<uint8_t>(GetFormat())];
+	
+	uint8_t* rtData = GetRawData();
+	const bool isFloat = stride == sizeof(float);
+
+	if (isFloat) {
+		float* imageData = stbi_loadf(filepath, &imageWidth, &imageHeight, nullptr, channels);
+		if (imageData) {
+			// Check if we require a resize to the RT's dimensions.
+			if (scaleToRT) {
+				uint16_t width = GetWidth();
+				uint16_t height = GetHeight();
+
+				// We can do a cool trick and skip any mediums and directly have the resized image be written into the RT's image buffer.
+				int resized = stbir_resize_float(imageData, imageWidth, imageHeight, channels * stride * imageWidth, reinterpret_cast<float*>(rtData), width, height, channels * stride * width, channels);
+				if (!resized) {
+					stbi_image_free(imageData);
+					return false;
+				}
+
+				stbi_image_free(imageData);
```

Comment:
Duplicate free call, just remove the above if statement and return `resized` instead